jetty-ee10/jetty-ee10-tests/jetty-ee10-test-http2-webapp/src/test/java/org/eclipse/jetty/test/webapp/StressHTTP2Test.java

```diff
@@ -0,0 +1,729 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+
+import jakarta.servlet.http.HttpServlet;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.Request;
+import org.eclipse.jetty.ee10.servlet.ServletContextHandler;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http2.client.HTTP2Client;
+import org.eclipse.jetty.http2.client.transport.HttpClientTransportOverHTTP2;
+import org.eclipse.jetty.http2.server.HTTP2CServerConnectionFactory;
+import org.eclipse.jetty.server.ConnectionFactory;
+import org.eclipse.jetty.server.HttpConfiguration;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.util.component.LifeCycle;
+import org.eclipse.jetty.util.thread.QueuedThreadPool;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+
+import static org.awaitility.Awaitility.await;
+
+@Tag("stress")
+@Timeout(value = 5, unit = TimeUnit.MINUTES)
+public class StressHTTP2Test
+{
+    private static final int N_THREADS = Runtime.getRuntime().availableProcessors();
+    private static final int ITERATIONS = 100;
+
+    private ExecutorService executorService;
+    private Server server;
+
+    @Test
+    public void testStressHTTP2WithAborts() throws Exception
+    {
+        List<Future<?>> futures = new ArrayList<>();
+        for (int i = 0; i < N_THREADS; i++)
+        {
+            Future<Object> f = executorService.submit(() ->
+            {
+                HTTP2Client http2Client = new HTTP2Client();
+                try (HttpClient httpClient = new HttpClient(new HttpClientTransportOverHTTP2(http2Client)))
+                {
+                    httpClient.start();
+
+                    for (int j = 0; j < ITERATIONS; j++)
+                    {
+                        CompletableFuture<Object> cf = new CompletableFuture<>();
+                        Request request = httpClient.newRequest(server.getURI());
+                        request.path("/")
+                            .method(HttpMethod.GET)
+                            .send(result ->
+                            {
+                                if (result.isSucceeded())
+                                    cf.complete(null);
+                                else
+                                    cf.completeExceptionally(result.getFailure());
+                            });
+
+                        if (j % (ITERATIONS / 10) == 0)
```

Comment:
Make it less regular.  Maybe a random?

Reply:
That would make the test more random, not more stressful.